pkg/reconciler/revision/resources/deploy.go

```diff
@@ -112,6 +131,18 @@ func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error)
 		}
 	}
 
+	// If concurrencyStateEndpoint is enabled, add the serviceAccountToken to QP via a projected volume
+	if cfg.Deployment.ConcurrencyStateEndpoint != "" {
+		podSpec.Volumes = append(podSpec.Volumes, varProjectedVolume)
+
+		for i, container := range podSpec.Containers {
```

Comment:
wondering if it makes more sense to do this in [makeQueueContainer](https://github.com/knative/serving/blob/60159d8e8d65afc0dc3d25b04724c2fa90d5e20b/pkg/reconciler/revision/resources/queue.go#L207) (called on the first line of this method), then we wouldn't have to loop containers to find the container we already added, and all the queue container bits would be in one place

Reply:
In that case, would we also want to move adding the volume to the podspec to the `BuildPodSpec` function or just leave that as is?

Reply:
we could do, but I don't feel particularly strongly about that. Mostly it just seems nice to avoid needing to loop containers to find the QP container when we made the QP container and added it to the list of containers in this exact method. An alternative if we wanted to keep all the logic in this method would be:

~~~~golang
queueContainer, err := makeQueueContainer(rev, cfg)
if err != nil {
	return nil, fmt.Errorf("failed to create queue-proxy container: %w", err)
}

var extraVolumes []corev1.Volume
if cfg.Deployment.ConcurrencyStateEndpoint != "" {
  queueContainer.VolumeMounts = append(queueContainer.VolumeMounts, varTokenVolumeMount)
  extraVolumes = append(extraVolumes, varTokenVolume)
}

podSpec := BuildPodSpec(rev, append(BuildUserContainers(rev), *queueContainer), cfg)
podSpec.Volumes = append(podSpec.Volumes, extraVolumes)
~~~~